src/__tests__/__utils__/checkAuth.test.js

```diff
@@ -0,0 +1,53 @@
+import { mount } from "enzyme";
+import React from "react";
+import configureStore from "redux-mock-store";
+import { Provider } from "react-redux";
+import { MemoryRouter } from "react-router-dom";
+import { checkAuth } from "../../utils/checkAuth";
+import { EditArticle } from "../../views/EditArticle";
+import { article } from "../__mocks__/testData";
+
+describe.only("should test the check authenticated component", () => {
+  const initialState = { article };
+  const mockStore = configureStore();
+  let wrapper;
+  let store;
+  let ConditionalComponent;
+  const nextUrl = "/articles/new";
+  beforeEach(() => {
+    const props = {
+      article,
+      match: {
+        params: {
+          slug: "hello-world"
+        }
+      },
+      response: "",
+      location: { pathname: nextUrl },
+      history: []
+    };
+    store = mockStore(initialState);
+    ConditionalComponent = checkAuth(EditArticle);
+    wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter>
+          <ConditionalComponent {...props} />
+        </MemoryRouter>
+      </Provider>
+    ).find("AuthenticatedComponent");
+  });
+  afterEach(() => {});
+  test("should redirect to next url if not logged in", () => {
+    expect(wrapper.props().history).toContain(`/sign_in?next=${nextUrl}`);
+  });
+
+  test("should return next component if the user is authenticated", () => {
+    // this test is giving me headaches
```

Comment:
I think this comment is inappropriate. Consider removing it!

Reply:
Done